client/state/sites/hooks/use-premium-global-styles.ts

```diff
@@ -0,0 +1,78 @@
+import { isEnabled } from '@automattic/calypso-config';
+import { useQuery } from 'react-query';
+import { useSelector } from 'react-redux';
+import wpcom from 'calypso/lib/wp';
+import { getSite } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+type GlobalStylesInfo = {
+	shouldLimitGlobalStyles: boolean;
+	globalStylesInUse: boolean;
+};
+
+type Site = {
+	isCreated: boolean;
+	id: number;
+};
+
+const NEW_SITE_GLOBAL_STYLES_INFO: GlobalStylesInfo = {
+	// The next line should be replaced with true once the Gating global styles feature is live.
+	// That will make all non-created sites to limit global styles.
+	shouldLimitGlobalStyles: isEnabled( 'limit-global-styles' ),
+	globalStylesInUse: false,
+};
+
+// While we are loading the Global Styles Info we can't assume that we should limit global styles, or we would be
+// showing notices for paid sites until we fetch the data from the server.
+const DEFAULT_GLOBAL_STYLES_INFO: GlobalStylesInfo = {
+	shouldLimitGlobalStyles: false,
+	globalStylesInUse: false,
+};
+
+const getGSInfoEndpointForSite = ( siteId: number ) => ( {
+	path: `sites/${ siteId }/global-styles-info`,
+	apiNamespace: 'wpcom/v2/',
+} );
+
+const mapSiteIdToSite = ( siteId: number | null ) => ( {
+	isCreated: !! siteId,
+	id: siteId ?? 0,
+} );
+
+const getGlobalStylesInfoForSite = ( site: Site ) => {
+	if ( ! site.isCreated ) {
+		return NEW_SITE_GLOBAL_STYLES_INFO;
+	}
+
+	const globalStylesInfoEndpoint = getGSInfoEndpointForSite( site.id );
+	return wpcom.req.get( globalStylesInfoEndpoint );
```

Comment:
Nitpick (non-blocking): I personally find this very hard to parse mentally. I prefer something like the below so we can effectively remove some functions like `mapSiteIdToSite` or `getGSInfoEndpointForSite` since they are not needed anywhere else.

```suggestion
const getGlobalStylesInfoForSite = ( siteId: number | null ) => {
	if ( ! siteId ) {
		return {
			shouldLimitGlobalStyles: isEnabled( 'limit-global-styles' ),
			globalStylesInUse: false,
		}
	}

	return wpcom.req.get( {
		path: `sites/${ siteId }/global-styles-info`,
		apiNamespace: 'wpcom/v2/',
	} );
```

Reply:
For me, it's the other way around XD. I personally find your suggestion much harder to follow since I need to parse everything that the code is doing instead of actually reading the code.

Reply:
It's a matter of getting used to it :D I found it confusing at the beginning too, but ended up paying off. I recommend you to take a look at this post: p7H4VZ-2av-p2

> Don’t wrap functions with new functions that opaquely pass along its parameters underneath. Be cautious of places where we’re adding one or two additional values onto the underlying function calls. These add indirection, meaning we have to jump through more code in order to understand them than is necessary.

> Don’t wrap APIs with more-constrained wrappers around those APIs. These tend to become quickly insufficient as people often want access to more of the underlying API than we initially assume they might

> Don’t split linear procedures into arbitrary linear sub-procedures to separate side-effects.